File: tests/MultiIndexTests.cpp
#include <gtest/gtest.h>

#include "clf/MultiIndex.hpp"

using namespace clf;

TEST(MultiIndexTests, BasicTest) {
  const std::size_t d = 5;
  std::vector<std::size_t> alpha(d);
  std::size_t order = 0;
  for( std::size_t i=0; i<d; ++i ) { 
    alpha[i] = rand()%13; 
    order += alpha[i];
  }

  MultiIndex ind(alpha);
  EXPECT_EQ(ind.Dimension(), d);
  EXPECT_EQ(ind.Order(), order);
}

File: tests/MultiIndexTests.py
"""! @brief Test the multi-index (clf::MultiIndex) python interface"""

import unittest

import PyCoupledLocalFunctions as clf
import random

class TestMultIndex(unittest.TestCase):
    """! Test the multi-index (clf::MultiIndex) python interface"""
    def test_construction(self):
        """! Test the multi-index construction"""
        dim = int(5)
        alpha = [None]*dim
        order = 0
        for i in range(dim): 
            alpha[i] = random.randint(0, 13)
            order += alpha[i]
        
        ind = clf.MultiIndex(alpha)
        self.assertEqual(ind.Dimension(), dim)
        self.assertEqual(ind.Order(), order)
      
